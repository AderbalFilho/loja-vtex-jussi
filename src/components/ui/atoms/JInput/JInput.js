import React, { useEffect, useState } from 'react';
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';

import JIcon from '../JIcon';

import './JInput.scss';

function JInput({
  classes,
  classesContainer,
  icon,
  iconClicked,
  internalValue,
  valueChange,
  ...rest
}) {
  const [value, setValue] = useState('');
  const className = classNames(
    'j-input__input',
    icon && 'j-input__input--with-icon',
    classes
  );
  const classNameContainer = classNames('j-input', classesContainer);

  useEffect(() => {
    setValue(internalValue);
  }, [internalValue]);

  function handleChange({ target }) {
    setValue(target.value);
    valueChange();
  }

  function handleIconClick() {
    iconClicked();
  }

  return (
    <div className={classNameContainer}>
      <input
        className={className}
        value={value}
        {...rest}
        onChange={handleChange}
      />
      {icon && (
        <JIcon
          className="j-input__icon"
          icon={icon}
          onClick={handleIconClick}
        />
      )}
    </div>
  );
}

JInput.defaultProps = {
  icon: '',
  iconClicked: () => {},
  internalValue: '',
  valueChange: () => {},
};

JInput.propTypes = {
  classes: PropTypes.string,
  classesContainer: PropTypes.string,
  icon: PropTypes.string,
  iconClicked: PropTypes.func,
  internalValue: PropTypes.string,
  valueChange: PropTypes.func,
};

export default JInput;
