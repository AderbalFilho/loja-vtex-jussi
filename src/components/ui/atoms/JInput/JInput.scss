@import 'styles/mixin';

.j-input {
  display: flex;
  width: 240px;
}

.j-input__input {
  @extend %text-paragraph;

  background: var(--white);
  border: 1px solid var(--grey);
  border-radius: 24px;
  color: var(--pink-jussi);
  padding: 8px 16px;
  width: 100%;

  &::placeholder {
    color: var(--pink-jussi);
  }

  &:focus {
    outline: none;
    border: 1px solid var(--grey);
  }
}

.j-input__input--with-icon {
  padding-right: 56px;
}

.j-input__icon {
  margin-left: -56px;
  padding: 8px 16px;
}
