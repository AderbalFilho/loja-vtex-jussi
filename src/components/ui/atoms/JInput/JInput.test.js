import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';

import JInput from './JInput';

describe('JInput', () => {
  it('matches snapshot', () => {
    const testRenderer = TestRenderer.create(<JInput type="text" />);
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });

  describe('with icon', () => {
    it('matches snapshot', () => {
      const testRenderer = TestRenderer.create(
        <JInput type="text" icon="search" />
      );
      const jsonComponent = testRenderer.toJSON();

      expect(jsonComponent).toMatchSnapshot();
    });

    it('fire event when icon clicked', () => {
      const onClick = jest.fn();
      const { getByAltText } = render(
        <JInput type="text" icon="search" iconClicked={onClick} />
      );

      fireEvent.click(getByAltText(/buscar/i));

      expect(onClick).toHaveBeenCalled();
    });
  });

  it('changes internal value', () => {
    const { container } = render(<JInput type="text" internalValue="Jüssi" />);
    const input = container.querySelector('.j-input__input');

    expect(input.value).toBe('Jüssi');
  });

  it('fire event when input text changed', () => {
    const onClick = jest.fn();
    const { getByPlaceholderText } = render(
      <JInput type="text" placeholder="Buscar" valueChange={onClick} />
    );

    fireEvent.change(getByPlaceholderText(/Buscar/i), {
      target: { value: 'Jüssi' },
    });

    expect(onClick).toHaveBeenCalled();
  });

  it('renders classes', () => {
    const { container } = render(<JInput classes="search-input" type="text" />);
    const input = container.querySelector('.j-input__input');

    expect(input.classList).toContain('search-input');
  });
});
