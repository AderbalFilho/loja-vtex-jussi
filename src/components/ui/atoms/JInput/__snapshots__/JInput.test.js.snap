// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JInput matches snapshot 1`] = `
<div
  className="j-input"
>
  <input
    className="j-input__input"
    onChange={[Function]}
    type="text"
    value=""
  />
  
</div>
`;

exports[`JInput with icon matches snapshot 1`] = `
<div
  className="j-input"
>
  <input
    className="j-input__input j-input__input--with-icon"
    onChange={[Function]}
    type="text"
    value=""
  />
  <img
    alt="buscar"
    className="j-input__icon"
    onClick={[Function]}
    src=""
  />
</div>
`;
